From 544e2a83c95263912917c9bc3fe63e1b7ee6eff2 Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Wed, 11 Apr 2018 11:11:53 +0300
Subject: [PATCH 06/11] Disable zram rule for now

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 data/80-udisks2.rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/80-udisks2.rules b/data/80-udisks2.rules
index f738e897..82221726 100644
--- a/data/80-udisks2.rules
+++ b/data/80-udisks2.rules
@@ -171,7 +171,7 @@ KERNEL=="sr*", ENV{ID_VENDOR}=="ASMT",ENV{ID_MODEL}=="VirtualCDROM", ENV{ID_FS_L
 ENV{ID_PART_TABLE_TYPE}=="dos", ENV{ID_PART_ENTRY_TYPE}=="0x0|0x17", ENV{ID_PART_ENTRY_NUMBER}=="1", ENV{ID_FS_TYPE}=="iso9660|udf", ENV{UDISKS_IGNORE}="0"
 
 # Zram devices setup
-KERNEL=="zram[0-9]", ENV{SYSTEMD_WANTS}="zram-setup@zram%n.service", TAG+="systemd"
+# KERNEL=="zram[0-9]", ENV{SYSTEMD_WANTS}="zram-setup@zram%n.service", TAG+="systemd"
 
 # Explicitly ignore ram block devices, they don't work with udev
 KERNEL=="ram*", ENV{UDISKS_IGNORE}="1"
-- 
2.17.1

