From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Franz-Josef Haider <franz.haider@jolla.com>
Date: Mon, 15 Mar 2021 13:29:59 +0000
Subject: [PATCH] Always mount filesystems using the UUID instead of label.

---
 src/udiskslinuxfilesystem.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/udiskslinuxfilesystem.c b/src/udiskslinuxfilesystem.c
index 02589fa230feac275ad6ef4f5a2943a2e38e3704..f9bd37ad1071c900574080dbe56c05931459580d 100644
--- a/src/udiskslinuxfilesystem.c
+++ b/src/udiskslinuxfilesystem.c
@@ -847,11 +847,16 @@ calculate_mount_point (UDisksDaemon  *daemon,
       *persistent = TRUE;
     }
 
+#if 1
+  (void)label;
+#else
   if (label != NULL && strlen (label) > 0)
     {
       mount_point = sanitize_mount_point (mount_dir, label);
     }
-  else if (uuid != NULL && strlen (uuid) > 0)
+  else
+#endif
+  if (uuid != NULL && strlen (uuid) > 0)
     {
       mount_point = sanitize_mount_point (mount_dir, uuid);
     }
