From d7b1f6298833b65e82635802f56ce3c38fb580ff Mon Sep 17 00:00:00 2001
From: Dmitry Gerasimov <d.gerasimov@omprussia.ru>
Date: Tue, 3 Nov 2020 17:17:46 +0300
Subject: [PATCH 12/13] Pass extra mount options to mount-sd service

---
 data/mount-sd@.service | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/data/mount-sd@.service b/data/mount-sd@.service
index e0ef0d07..97cad224 100644
--- a/data/mount-sd@.service
+++ b/data/mount-sd@.service
@@ -8,5 +8,6 @@ Conflicts=rescue.target actdead.target factory-test.target
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/usr/bin/udisksctl-user mount -b /dev/%i
+EnvironmentFile=-/var/lib/environment/udisks2/*.conf
+ExecStart=/usr/bin/udisksctl-user mount $UDISKS2_MOUNT_OPTIONS -b /dev/%i
 ExecStop=/usr/bin/udisksctl unmount -b /dev/%i
-- 
2.25.1

