From 4273858e5bb8832c8872ddd7f019149e675d58b0 Mon Sep 17 00:00:00 2001
From: Franz-Josef Haider <franz.haider@jolla.com>
Date: Mon, 15 Mar 2021 13:29:59 +0000
Subject: [PATCH 13/13] Always mount filesystems using the UUID instead of
 label.

---
 src/udiskslinuxfilesystem.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/udiskslinuxfilesystem.c b/src/udiskslinuxfilesystem.c
index 60cc254c..81c370f5 100644
--- a/src/udiskslinuxfilesystem.c
+++ b/src/udiskslinuxfilesystem.c
@@ -705,6 +705,9 @@ calculate_mount_point (UDisksDaemon  *daemon,
    *
    *       See http://en.wikipedia.org/wiki/UTF-8 for details.
    */
+#if 1
+  (void)label;
+#else
   if (label != NULL && strlen (label) > 0)
     {
       str = g_string_new (NULL);
@@ -721,7 +724,9 @@ calculate_mount_point (UDisksDaemon  *daemon,
       mount_point = g_string_free (str, FALSE);
       g_free (s);
     }
-  else if (uuid != NULL && strlen (uuid) > 0)
+  else
+#endif
+  if (uuid != NULL && strlen (uuid) > 0)
     {
       str = g_string_new (NULL);
       g_string_append_printf (str, "%s/", mount_dir);
-- 
2.25.1

